rotors_gazebo_plugins/src/gazebo_fw_dynamics_plugin.cpp

```diff
@@ -0,0 +1,406 @@
+
+  // ============================================ //
+  // ==== WIND SPEED MSG SETUP (ROS->GAZEBO) ==== //
+  // ============================================ //
+
+  // Wind speed subscriber (Gazebo).
+  wind_speed_sub_ =
+      node_handle_->Subscribe("~/" + namespace_ + "/" + wind_speed_sub_topic_,
+                              &GazeboFwDynamicsPlugin::WindSpeedCallback,
+                              this);
+
+  connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" +
+                                                wind_speed_sub_topic_);
+  connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" +
+                                                   wind_speed_sub_topic_);
+  connect_ros_to_gazebo_topic_msg.set_msgtype(
+      gz_std_msgs::ConnectRosToGazeboTopic::WIND_SPEED);
+  gz_connect_ros_to_gazebo_topic_pub->Publish(connect_ros_to_gazebo_topic_msg,
+                                              true);
+
+  // If we are using a joystick for control inputs then subscribe to the
+  // RollPitchYawrateThrust msgs, otherwise subscribe to the Actuator msgs.
+  if (is_input_joystick_) {
+    // ========================================================= //
+    // === ROLL_PITCH_YAWRATE_THRUST MSG SETUP (ROS->GAZEBO) === //
+    // ========================================================= //
+    roll_pitch_yawrate_thrust_sub_ =
+        node_handle_->Subscribe("~/" + namespace_ + "/" +
+            roll_pitch_yawrate_thrust_sub_topic_,
+            &GazeboFwDynamicsPlugin::RollPitchYawrateThrustCallback, this);
+
+    connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" +
+        roll_pitch_yawrate_thrust_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" +
+        roll_pitch_yawrate_thrust_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_msgtype(
+        gz_std_msgs::ConnectRosToGazeboTopic::ROLL_PITCH_YAWRATE_THRUST);
+  } else {
+    // ============================================ //
+    // ===== ACTUATORS MSG SETUP (ROS->GAZEBO) ==== //
+    // ============================================ //
+
+    actuators_sub_ =
+        node_handle_->Subscribe("~/" + namespace_ + "/" + actuators_sub_topic_,
+                                &GazeboFwDynamicsPlugin::ActuatorsCallback,
+                                this);
+
+    connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" +
+                                                  actuators_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" +
+                                                     actuators_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_msgtype(
+        gz_std_msgs::ConnectRosToGazeboTopic::ACTUATORS);
+  }
+
+  gz_connect_ros_to_gazebo_topic_pub->Publish(connect_ros_to_gazebo_topic_msg,
+                                              true);
+}
+
+void GazeboFwDynamicsPlugin::ActuatorsCallback(
+    GzActuatorsMsgPtr &actuators_msg) {
```

Comment:
'GzActuatorsMsgPtr' is already typedef'ed as 'const boost::shared_ptr<const gz_sensor_msgs::Actuators>' in the header for shorter notation, much like it was recently done in other plugins.